force-app/main/default/classes/RollupRecord.cls

```diff
@@ -0,0 +1,121 @@
+public with sharing class RollupRecord {
+    public Id recordId;
+    @TestVisible
+    private Integer presentCount;
+    @TestVisible
+    private Integer absentCount;
+    @TestVisible
+    private Integer consecutiveCount;
+    @TestVisible
+    private Date lastServiceDate;
+    @TestVisible
+    private Integer totalRecords;
+    @TestVisible
+    private Integer totalProcessed;
+    @TestVisible
+    private SObjectField numAbsentField;
+    @TestVisible
+    private SObjectField numPresentField;
+    @TestVisible
+    private SObjectField consecutiveAbsencesField;
+    @TestVisible
+    private SObjectField lastServiceDateField;
+
+    public RollupRecord(Id recordId) {
+        this.recordId = recordId;
+        this.totalProcessed = 0;
+    }
+
+    public Boolean getIsProcessed() {
+        return totalRecords == null || totalProcessed >= totalRecords;
+    }
+
+    public void setTotalRecords(Integer totalRecords) {
+        this.totalRecords = totalRecords;
+    }
+
+    public void rollupFields(
+        SObjectField numAbsentField,
+        SObjectField numPresentField,
+        SObjectField consecutiveAbsencesField,
+        SObjectField lastServiceDateField
+    ) {
+        this.numAbsentField = numAbsentField;
+        this.numPresentField = numPresentField;
+        this.consecutiveAbsencesField = consecutiveAbsencesField;
+        this.lastServiceDateField = lastServiceDateField;
+    }
+
+    public void incrementProcessed() {
+        totalProcessed++;
+    }
+
+    public void incrementPresentCount() {
+        if (numPresentField == null) {
+            return;
+        }
+
+        presentCount = presentCount == null ? 1 : presentCount + 1;
+        absentCount = absentCount == null ? 0 : absentCount;
+        consecutiveCount = 0;
+    }
+
+    public void incrementAbsentCount() {
+        if (numAbsentField == null) {
+            return;
+        }
+
+        presentCount = presentCount == null ? 0 : presentCount;
+        absentCount = absentCount == null ? 1 : absentCount + 1;
+        consecutiveCount = consecutiveCount == null ? 1 : consecutiveCount + 1;
```

Comment:
This will likely give an incorrect count if the batches are run out of order.  Might have to persist a list and calculate at the end.